contracts/Vaults/CrabVault.sol

```diff
@@ -0,0 +1,513 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+
+import "../utils/proxy/solidity-0.8.0/ProxyReentrancyGuard.sol";
+import "../utils/proxy/solidity-0.8.0/ProxyOwned.sol";
+
+import "../interfaces/IPositionalMarket.sol";
+import "../interfaces/IStakingThales.sol";
+
+import "../RangedMarkets/RangedMarketsAMM.sol";
```

Comment:
importing the whole RangedMarketsAMM is not the best idea here as the contract size will then breach etherscan verification limits. 
Its better to create an interface IRangedMarketsAMM that will expose only the method which is used in this contract.